### DISCUSSION TITLE
Add menu to the Tab control

### DISCUSSION THREAD

**Discussion by hexwit** [mantinedev/mantine#1805] [Feature requests]
It would be nice to have possibility to have menu in the Tab component. 
Currently it is possible to do via 'control' property, but it is awkward.

**Comment by rtivital** [comment]
It will not be accessible – tab button will have 2 different roles, we will not add this as a part of the library

**Reply by hexwit** [reply]
So the only way is to customise 'control' property?

**Reply by rtivital** [reply]
Yes

**Reply by hexwit** [reply]
@rtivital Could you suggest what would be correct way to incorporate <Menu> into <Tab> for the V5? 
I tried children, I tried rightSection. I tried provide <span> for <Menu.Target>.

Everywhere I see "Warning: validateDOMNesting(...): <button> cannot appear as a descendant of <button>.". 

Thanks.

**Reply by rtivital** [reply]
The only way I can think of is that you can render an element that looks the same way as a tab, but not the `Tabs.Tab` component itself.

**Reply by hexwit** [reply]
That would mean I should implement whole Tab component from scratch for the fair standard functionality. 
In V4 there was a 'control' property. It will not be available anymore?

**Reply by rtivital** [reply]
`control` prop was replaced with Menu.Target `children`

**Reply by hexwit** [reply]
Got it. Is it possible to have controlled menu that will not render button internally? 
If I control the menu - button is not needed, and it is even not displayed.

**Reply by rtivital** [reply]
No, Menu is attached to target element. If you do not provide a target, then the Menu will not work, as the element that should be used to calculate dropdown position will not exist.

**Reply by hexwit** [reply]
Even if I provide span as a target, somewhere under the hood, button is used.
That is a pretty frustrating situation. Such a simple standard task and cannot be implemented.
 
Is it possible to return 'control' property that will replace whole tab component like that was in V4? 
Because 'children' currently works differently.

**Reply by rtivital** [reply]
No, control prop will not work from 5.0

**Reply by hexwit** [reply]
Seems to be popover can be attached and customized to have a look like a menu, and be in rightSection. That should be easier than implement Tabs from scratch.
